bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BookKeeperBuildersTest.java

```diff
@@ -340,8 +337,12 @@ public void testOpenLedgerNoRecovery() throws Exception {
 
         ledgerMetadata.getEnsembles().values().forEach(bookieAddressList -> {
             bookieAddressList.forEach(bookieAddress -> {
-                registerMockEntryForRead(ledgerId, BookieProtocol.LAST_ADD_CONFIRMED, bookieAddress, entryData, -1);
-                registerMockEntryForRead(ledgerId, 0, bookieAddress, entryData, -1);
+                try {
+                    registerMockEntryForRead(ledgerId, BookieProtocol.LAST_ADD_CONFIRMED, bookieAddress, entryData, -1);
+                    registerMockEntryForRead(ledgerId, 0, bookieAddress, entryData, -1);
```

Comment:
Last thing: why registerMockEntryForRead can throw a BKException? Can't we throw a cchecked exception? Or as last resort you can ise @SneakyThrows.

Reply:
good idea. when the bookie is down, the write to it will throw NoBookieAvalibleException to trigger ensemble change. I tried @sneakythrows, but fail. I can't add @sneakythrows in `registerMockEntryForRead`, because the exception is needed to response failure to request. I failed to add @sneakythrows(...BKException) on methods who call registerMockEntryForRead(such as `testOpenLedgerRecovery`), which causes compile error, "must be caught ...", although it should be compiled with no error.